pkg/filter/filter.go

```diff
@@ -108,24 +119,44 @@ func JoinResourceTypeMatchers(ms ...ResourceTypeMatcher) ResourceTypeMatcher {
 
 // ResourceRequirement allows specifying a resource requirement by type and/or name
 type ResourceRequirement struct {
+	// Provides the Name of the resource object
 	v1.LocalObjectReference `json:",inline,omitempty"`
+	// Provides the Group, Version, and Resource values (GVR)
 	ResourceTypeRequirement `json:",inline,omitempty"`
+	// Specifies a set of label requirements to be used as filters for matches
+	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
-// Matches returns true if the specified resource name/GVR matches the requirement
-func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource) bool {
-	// If the requirement does not specify a resource name - only check the
-	// ResourceTypeRequirement i.e. GVR match
-	if r.LocalObjectReference.Name == "" {
-		return r.ResourceTypeRequirement.Matches(gvr)
+// Matches returns true if the specified resource name/GVR/labels matches the requirement
+func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource, resourceLabels map[string]string) bool {
+	// If ResourceRequirement Name/GVR value of empty string matches on any specified in parameters.
+	// If ResourceRequirement Name/GVR value specified, exact match on parameter is required.
+	if !(matches(r.Name, name) && r.ResourceTypeRequirement.Matches(gvr)) {
+		return false
 	}
-	return matches(r.Name, name) && r.ResourceTypeRequirement.Matches(gvr)
+	if len(resourceLabels) == 0 && len(r.MatchExpressions) == 0 && len(r.MatchLabels) == 0 {
```

Comment:
@vkamra, @tdmanv We need a policy clarification/decision here please.  If there is no LabelSelector in the requirement, should we / can we return a true match regardless what the labels are on the object.

@tylerje Another way of asking this is, what will an empty LabelSelector return from sel.Matches below?

Reply:
@ewhamilton - I think that makes sense. If the user didn't specify labels as a requirement - the `resourceLabels` should be ignored.

Reply:
@tyler In the interest of clarity and conceptual alignment, we can remove the `len(resourceLabels) == 0 && ` term from line 137 and update the comment on line 138.

This makes it clearer that:
* If there is no required Name (r.Name == ""), any name (including none) matches
* If there is no required Group (r.Group == ""), any group (including none) matches
* If there is no required Version (r.Version == ""), any version (including none) matches
* If there is no required Resource (r.Version == ""), any resource (including none) matches
* If there is no required LabelSelector (r.LabelSelector == LabelSelector{}), any label (including none) matches